CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -72,19 +72,42 @@ namespace CefSharp
             CefRefPtr<CefClient>& client, CefBrowserSettings& settings, bool* no_javascript_access)
         {
             ILifeSpanHandler^ handler = _browserControl->LifeSpanHandler;
+			IWebBrowser^ newBrowser = nullptr;
 
             if (handler == nullptr)
             {
                 return false;
             }
 
-            return handler->OnBeforePopup(_browserControl, gcnew CefFrameWrapper(frame, _browserControl->BrowserAdapter), StringUtils::ToClr(target_url),
-                windowInfo.x, windowInfo.y, windowInfo.width, windowInfo.height, *no_javascript_access);
+            bool returnVal = handler->OnBeforePopup(_browserControl, gcnew CefFrameWrapper(frame, _browserControl->BrowserAdapter), StringUtils::ToClr(target_url),
+				windowInfo.x, windowInfo.y, windowInfo.width, windowInfo.height, *no_javascript_access, newBrowser);
+
+			IWebBrowserInternal^ newBrowserInternal = dynamic_cast<IWebBrowserInternal^>(newBrowser);
```

Comment:
Without looking at this too closely, it appears you've cast more times than is required. What's stopping you from directly casting to `IRenderWebBrowser`?

Should probably check for `newBrowser` to be null before attempting anything. Also code needs comments to document the behavior.

You may not even need to cast as it doesn't appear your using the result.